src/Routing.purs

```diff
@@ -25,10 +25,13 @@ foreign import decodeURIComponent :: String -> String
 foreign import hashChanged """
 function hashChanged(handler) {
   return function() {
-    var currentHash = document.location.hash;
-    handler("")(currentHash)();
+    var getHash = function() {return document.location.href.split('#')[1] || '';};
+    window['__oldHash'] = "";
```

Comment:
Do you think it might be better to keep this as a private local variable instead? Could define `hashChanged` as something like:

``` js
var hashChanged = (function () {
  var oldHash = "";
  return function (handler) {
    ...
  }
}());
```